lib/vanagon/project.rb

```diff
@@ -476,5 +477,36 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json file which lists all of the dependant components of the project
+    def generate_dependencies_info
+      project_components = @components.map do |component|
+        components_hash = component.options
+        [:version, :url].each do |key|
+          if val = component.send(key)
+            components_hash[key] = val
+          end
+        end
+        [
+          component.name,
+          components_hash,
+        ]
+      end.to_h
+    end
+
+    def generate_manifest_json
+      build_time = Time.now
+      manifest = {
+        "packaging_type" => {
+          "vanagon" => Gem.loaded_specs["vanagon"].version.to_s,
+        },
+        "git_sha" => @version,
+        "components" => generate_dependencies_info,
+        "build_time" => build_time.rfc2822,
```

Comment:
Why not just store `build_time` as RFC2822? Also, maybe consider ISO8601 instead of RFC2822:

```ruby
>> Time.now.iso8601
=> "2017-07-03T23:31:42-07:00"
```
(RFC2822 is for email timestamp headers. ISO8601 is a legit worldwide standard.)